armi/materials/__init__.py

```diff
@@ -120,30 +133,42 @@ def resolveMaterialClassByName(name: str, namespaceOrder: List[str] = None):
     Find the first material class that matches a name in an ordered namespace.
 
     Names can either be fully resolved class paths (e.g. ``armi.materials.uZr:UZr``)
-    or simple class names (e.g. ``UZr``). In the latter case, the ``CONF_MATERIAL_NAMESPACE_ORDER``
-    setting to allows users to choose which particular material of a common name (like UO2 or HT9)
-    gets used.
+    or simple class names (e.g. ``UZr``). In the latter case, the
+    ``CONF_MATERIAL_NAMESPACE_ORDER`` setting to allows users to choose which
+    particular material of a common name (like UO2 or HT9) gets used.
 
     Input files usually specify a material like UO2. Which particular implementation
     gets used (Framework's UO2 vs. a user plugins UO2 vs. the Kentucky Transportation
     Cabinet's UO2) is up to the user at runtime.
 
-    .. impl:: Material collections are defined with an order of precedence in the case of duplicates.
+    .. impl:: Material collections are defined with an order of precedence in the case
+        of duplicates.
         :id: I_ARMI_MAT_ORDER
         :implements: R_ARMI_MAT_ORDER
 
+        During the runtime of an ARMI application, but particularly during the
+        construction of the reactor in memory, materials will be identified by name. At
+        that point, this code is called to search for that material name. The search
+        goes through the ordered list of Python namespaces provided to search for the
+        name. The first time an instance of that material is found, it is returned. In
+        this way, the first items in the material namespace list take precedence.
+
+        When a meterial name is pased to this function, it might be something simple
+        like the string ``"UO2"``. But it could be much more specific, like
+        ``armi.materials.uraniumOxide:UO2``.
```

Comment:
```suggestion
        During the runtime of an ARMI application, but particularly during the
        construction of the reactor in memory, materials will be requested from the 
        materials plugin by name. At that point, this code is called to search for that 
        material name. The search goes through the ordered list of Python 
        namespaces provided to search for the name. The first time an instance of 
        that material is found, it is returned. In this way, the first items in the material 
        namespace list take precedence.
        
        When a material name is passed to this function, it may be either a simple 
        name like the string ``"UO2"`` or it may be much more specific, like
        ``armi.materials.uraniumOxide:UO2``.
```
Rephrase second paragraph as needed, I think it would be good to provide a more complete description of the options available. Based on my reading of the other portions of the docstring, these are the two options.